File: 13_student_grade_manager/cpp/student_grade_manager.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

struct Student {
    string name;
    double grade;
};

int main() {
    vector<Student> students;

    while (true) {
        cout << "\n1. Add Student\n2. View Students\n3. Show Average\n4. Exit\nChoice: ";
        int choice;
        cin >> choice;

        if (choice == 1) {
            Student s;
            cout << "Student name: ";
            cin >> s.name;
            cout << "Grade: ";
            cin >> s.grade;
            students.push_back(s);
            cout << "Student added.\n";
        }
        else if (choice == 2) {
            if (students.empty())
                cout << "No students available.\n";
            for (auto& s : students)
                cout << s.name << ": " << s.grade << endl;
        }
        else if (choice == 3) {
            if (!students.empty()) {
                double sum = 0;
                for (auto& s : students)
                    sum += s.grade;
                cout << "Class Average: " << sum / students.size() << endl;
            } else {
                cout << "No grades to calculate.\n";
            }
        }
        else if (choice == 4) {
            break;
        }
        else {
            cout << "Invalid choice.\n";
        }
    }
    return 0;
}

File: 13_student_grade_manager/python/student_grade_manager.py
students = []

while True:
    print("\n1. Add Student")
    print("2. View Students")
    print("3. Show Average")
    print("4. Exit")

    choice = input("Choose option: ")

    if choice == "1":
        name = input("Student name: ")
        grade = float(input("Grade: "))
        students.append((name, grade))
        print("Student added.")

    elif choice == "2":
        if not students:
            print("No students available.")
        for name, grade in students:
            print(f"{name}: {grade}")

    elif choice == "3":
        if students:
            avg = sum(g for _, g in students) / len(students)
            print("Class Average:", avg)
        else:
            print("No grades to calculate.")

    elif choice == "4":
        break

    else:
        print("Invalid choice.")
